cmd/agent/DEVELOPERS.md

```diff
@@ -3,6 +3,18 @@
 This document provides relevant instructions for maintainers of the Grafana
 Agent.
 
+## Branching Strategy
+
+The Grafana Agent's branching strategy has shifted with the release of v0.19.0
+and is moving to a main always releasable approach. This will allow us to better control
+the release cadence and ensure that the project has regular releases. 
+
+For contributors, any large change should be made in a `dev.*` branch. These branches allow work
+on large features without crowding main. If you would like to work on a large feature let @rfratto or
+@mattdurham know, and we can create a branch for you.
+
+Small features can still be via direct PRs.
```

Comment:
Maybe move this above? I don't want the first thing for potential contributors to see is the dev branch spiel, since it might scare people off. 

Can we also ask that if it's going to take them effort to implement, we'd recommend asking / opening an issue first so they don't have to waste their time for something that gets turned down?